Enhancement/forms

NeoVier opened the pull request:
## What issue does this PR close
Closes #585, closes #567

## Changes Proposed ( a list of new changes introduced by this PR)
- [x] Add a new `Form` module, responsible for all forms in our app. It's pretty complex, but it's pretty well documented, and there's also more documentation on [hecrj/composable-form](https://package.elm-lang.org/packages/hecrj/composable-form/latest/), which was a great inspiration for this module, so if you want to understand how it all works and how to use it, check those out!
- [x] Add new `Form.*` modules, such as `Form.Text`. These are responsible for each kind of field we can have in our app. They're pretty much the same as the previous `View.Form.*` modules, but refactored to work with the new `Form` module
- [x] Add `UR.addChild` and `UR.fromChild` helpers. With these, we can have components that use `UpdateResult` in their `update` function, and have a better API for getting results from these components (`Form` is one example that already uses `UpdateResult` on it's `update` function).
- [x] Replace existing forms with the new api
- [x] Remove the old `View.Form.*` modules. These shouldn't be needed anymore with the new form field modules.
- [x] Add documentation about the new Form API and the new Fields on elm-book

## How to test ( a list of instructions on how to test this PR)

Since this changes every single form in our app, it's a good idea to run through the app and give it a complete test anywhere there is a form, checking if everything works. Here are some things that are trickier and might require some more attention:
- Masks on text fields
- Different input types that show different keyboards on mobile (e.g. phone number inputs show a phone number keyboard, numeric inputs show a numerical keyboard, etc.)
- The rich text editor
- The file uploader
- The user selector
- The date selector
- The dropdown selector

Also, now we have something called elm-book! It's a visual guide to all our components (only form components are included for now), and you can access it at https://cambiatus-elm-book.netlify.app/, or on the same area where you can find our regular netlify preview of the app (scroll all the way down this page, near where you can type a comment). Make sure the language used and all of the examples are coherent. You can use those examples to more easily test our form components!

NeoVier commented on src/elm/Page/FormPlayground.elm:
```diff
@@ -0,0 +1,292 @@
+update : Msg -> Model -> LoggedIn.Model -> UpdateResult
+update msg model _ =
+    case msg of
+        GotUserFormMsg subMsg ->
+            Form.update subMsg model.userFormModel
+                |> UR.fromChild
+                    (\userFormModel -> { model | userFormModel = userFormModel })
+                    GotUserFormMsg
+                    SubmittedUser
+                    model
```
@lucca65 this is what I was talking about having a component using `UpdateResult`. In this case, whenever a valid `User` is submitted, the `SubmittedUser` message is fired, and then the parent (this page) can do whatever it wants with the user

lucca65 commented:
🏋️

lucca65 reviewed (changes requested):
### No translation and no outline on the whole field
<img width="1297" alt="Captura de Tela 2022-01-17 às 18 16 21" src="https://user-images.githubusercontent.com/1082127/149837823-bdc556fb-572b-4567-9ce8-406246b36a4e.png">

### Tooltip goes on top of the calendar 
Seem in action form, the tool tip in question is on the automatic verification toggle

<img width="1269" alt="Captura de Tela 2022-01-17 às 18 45 16" src="https://user-images.githubusercontent.com/1082127/149840248-48e928d0-b999-4c05-acd1-b299039b622e.png">

NeoVier commented:
> no outline on the whole field

The "blurred" outline only shows up when you're focusing the field. That's how we can distinguish between focused and unfocused fields. Fields with errors show up with a red border (pictured in your image). If that's not the outline you meant, can you explain a little better?

---

Other than that, both issues should be fixed! The missing translation only happened because I must've worked on the transfer page before coming up with `Form.Validate`. The `Form.Validate` module offers some nice API to work with form validation, and has automatic translation, so it should be hard to happen again in the future! I've changed the Transfer page to use it

lucca65 commented:
> The "blurred" outline only shows up when you're focusing the field. That's how we can distinguish between focused and unfocused fields. Fields with errors show up with a red border (pictured in your image). If that's not the outline you meant, can you explain a little better?

I mean, the CMB add-on with blue background on the input doesn't get highlighted in red

NeoVier commented:
Ok, fixed!

lucca65 commented on src/elm/Page/Register.elm:
```diff
@@ -27,12 +29,10 @@ import Result
-import View.Form.Checkbox as Checkbox
-import View.Form.Input as Input
 
 
```
Same thing as the other page, something's up with the validation trigger, should happen only if something has changed, at least on the first time

https://user-images.githubusercontent.com/1082127/150032668-22dd2736-5976-4b4e-a7ce-19e8171711a6.mov

lucca65 commented on src/elm/Page/Shop/Editor.elm:
```diff
@@ -65,21 +60,6 @@ initUpdate productId loggedIn =
-                |> MarkdownEditor.withSubscription form.description
-
-
-
```
Same thing is happening on this page, if I focus in, it gives an error. On `title` and `price`

lucca65 commented on src/elm/Select/Select.elm:
```diff
@@ -2,6 +2,7 @@ module Select.Select exposing (view)
 
```
Maybe this module can be inside `/src/elm/Components`?

lucca65 commented on src/elm/Profile/Contact.elm:
```diff
@@ -999,25 +1002,37 @@ viewFlagsSelect { t } basic =
-            |> Input.withType Input.Telephone
-            |> Input.toHtml
-        ]
+    Form.Text.init
```
On the WhatsApp input, I can add letters to the number

lucca65 commented on src/elm/Profile/Contact.elm:
```diff
@@ -984,7 +987,7 @@ viewFlagsSelect { t } basic =
             , if basic.showFlags then
                 div
-                    [ class "absolute form-input -mx-px inset-x-0 top-0 space-y-4 z-50 h-44 overflow-auto"
+                    [ class "absolute input -mx-px inset-x-0 top-0 space-y-4 z-50 h-44 overflow-auto"
```
There is this weird bug happening....

Its kinda complicate to replicate, maybe there is a simpler route but here we go:

1. edit your contact options
2. Make sure you have something filled for WhatsApp, if not add one and reload the page for a fresh state
2. delete one contact, say telegram
3. fill telegram again back
4. Submit the page
5. go into edit mode again
6. notice how the WhatsApp number is no gone

Here is a demo:
https://user-images.githubusercontent.com/1082127/150033530-811055b5-ccf5-4933-b2a6-44a6dfe1abe8.mov

NeoVier replied:
Ok, that was a wild ride! Apparently, it was because we were removing the mask of the phone inputs before validating, but now the mask is automatically removed. So removing the mask was causing the input to be empty, which is the same as deleting it. Should be fixed now!

lucca65 replied:
nice bro!

lucca65 reviewed (changes requested):
🥵 I'm sweating, but finally managed to review it all!

NeoVier commented on src/elm/Page/Community/New.elm:
```diff
@@ -58,89 +50,240 @@ init _ =
+        |> Form.with
+            (Form.Toggle.init
+                { label = text <| t "settings.community_info.fields.invitation"
+                , id = "require-invitation-toggle"
```
Should we do this for every toggle or just this one in specific? We could add borders on large-enough screens

NeoVier replied:
Uh, I think that kind of input doesn't really fit well into this use case... I think that fits better when there are multiple options (not just yes/no). You can even see the little radio circle thing in this example (taken from CleanMyMacX):
<img width="1173" alt="image" src="https://user-images.githubusercontent.com/39280468/150219351-bb8c007b-1a03-49fa-993b-2e500fa03757.png">


I'll add borders around this specific one, and we can do the same for the rest if we feel like we need it in the future, does that seem ok?

lucca65 replied:
> Uh, I think that kind of input doesn't really fit well into this use case... I think that feels more like a radio button

Oh I only meant the border, not the option to choose between multiple options!

> I'll add borders around this specific one, and we can do the same for the rest if we feel like we need it in the future, does that seem ok?

Sounds perfect!

NeoVier replied:
I've added the border around the toggle, can you check if that's what you meant?

lucca65 replied:
Nice, looks awesome!

lucca65 commented:
@NeoVier do you think we can fix this on this PR? If it is simple enough

NeoVier commented:
> 🥵 I'm sweating, but finally managed to review it all!

It was actually way quicker than I thought it would be 🔥 

---

About #675, is that with this PR? I can't seem to replicate it, and looking at the code, everything in that page already uses the Feedback component - I might be way ahead of you 🏃

lucca65 commented:
> everything in that page already uses the Feedback component - I might be way ahead of you 🏃

Damn, that's what I'm talking about 🔥

NeoVier commented:
Ok, I changed the form's validation logic, it now works like this:

0) Field starts empty - no error message
1) User starts filling field - no error message
2) User clicks out of field - error message is shown (if there is an error)
3) User clicks back in on the field - error message is still shown
4) Go back to `1` (the error is hidden as the user is typing)

Empty fields won't show errors, unless when the user tries to submit the form. Also, I added a new `ValidationStrategy` thing, where we can configure each field separately to be validated on blur (lose focus) or only when the form is submitted. It seems to be especially useful for multiple-user-selections (e.g. on the action editor page), because (with the current api and how it works) the user picker automatically loses focus when the user selects a profile

---

I'm still not 100% sure this is how it should work, let me know if you have any suggestions! Form validation seems to be a pretty complex topic... here's a good thread on it: https://ux.stackexchange.com/questions/48268/validate-on-blur-or-keypress

NeoVier commented:
@lucca65 everything should be addressed! The only things left for now are:
- [Check if the Toggle border is what you meant](https://github.com/cambiatus/frontend/pull/658#discussion_r788225516)
- [Check if you have any comments/concerns on the way we deal with validation](https://github.com/cambiatus/frontend/pull/658#issuecomment-1016952509)

lucca65 commented:
> where we can configure each field separately to be validated on blur (lose focus) or only when the form is submitted

I think this is ideal, in a lot of cases, say the validation that shows an error when you fill a bigger amount than you have a balance for on transfer form, we should validate on lose focus, but others we can do on the submit.

lucca65 reviewed (changes requested):
Bro, found one more problem...

I'm not sure what is the context for this but check it out, the objectives section is misaligned:
<img width="1506" alt="Captura de Tela 2022-01-20 às 11 10 33" src="https://user-images.githubusercontent.com/1082127/150354673-fa36a654-277b-4d33-854d-d44a0fc40172.png">

NeoVier commented:
Uh that's weird... It looks normal to me, no matter the browser:
<img width="1294" alt="image" src="https://user-images.githubusercontent.com/39280468/150403506-ba6edd77-59d4-4e22-a151-f073dd2b3c6b.png">

Does it happen consistently if you refresh the page? I also noticed there isn't the "VOCÊ" circle under "Nossos Apoiadores", that's also pretty weird...

If it does happen consistently, can you try force reloading the page, or try it in an anonymous tab? It could be some caching issues

lucca65 commented:
> If it does happen consistently, can you try force reloading the page, or try it in an anonymous tab? It could be some caching issues

Yeah not consistent at all... I can't replicate it no more

> I also noticed there isn't the "VOCÊ" circle under "Nossos Apoiadores", that's also pretty weird...

It shows even if there is a profile picture

lucca65 reviewed (approved):
🏋️ 

Amazing bro, congratulations on the work,, the PR itself is kind of old, one month of work if we consider we stopped during the end of last year, but the desire to pursue this standard way of dealing with forms is from 24 Jun 2021, its been a ride and kudos to you for the hard work!

NeoVier commented:
> Bro, for the sake of simplicity, can you open a new issue for this "VOCÊ" problem?

That's how it was specified on the designs... Do you think we should change it? If so, I can open a new issue

---

Yeah, this was a lot of work, but I feel like I learned a fair bit, and I'm really satisfied with the results. I think this will definitely help us a lot in the long run 💪

lucca65 commented:
bro, don't this PR also fixes this one? https://github.com/cambiatus/frontend/issues/567

NeoVier commented:
I honestly can't tell 😬

IIRC that bug was only happening on iOS safari, and I don't have an iPhone I can test with... If you do, please test it out and see if it still happens! I think it should probably be fixed, as I changed a lot how inputs and their borders are treated, but I never figured out what was causing that behavior

juramos-2020 commented:
@NeoVier 

**- Login | mobile :** ao digitar o PIN no segundo campo do login, o primeiro campo é apagado e fica sinalizado com a borda vermelha, como se o campo tivesse sido preenchido de forma errada.

![PIN_Login](https://user-images.githubusercontent.com/67649886/150575527-cb110acf-9c27-42f2-a55c-4d97a8f1ff80.jpeg)


**- Login | desktop :**  após inserir as 12 palavras, a tela seguinte já aparece com o 1º campo do PIN preenchido. Preencho o segundo campo e consigo entrar na página normal. Só não sei se esse comportamento de preencher automaticamente o primeiro campo é certo. Isso nunca tinha acontecido antes, eu sempre tive que digitar meu PIN nos dois campos.

![image](https://user-images.githubusercontent.com/67649886/150576391-d3b0055a-0919-4400-9961-4e26c4e2de2b.png)

juramos-2020 commented:
**- Perfil | Mobile:**

- O icone de edição do perfil não está funcionando. Só consegui acessar a página de edição pois atualizei a página. 
- O ícone de "?" do KYC pode seguir a cor padrão dos botões (#F99D33)
- Adicionar um espaçamento de 8px entre o botão "Excluir" do KYC e o texto
- A palavra "Excluir" está repetida dentro do botão
- A digitação dentro do campo de biografia não está funcionando corretamente
- A palavra "adicionar" está repetida dentro do botão



<img width="1726" alt="Perfil2" src="https://user-images.githubusercontent.com/67649886/150583212-90fdd6da-cc03-4c94-a9b2-205430a17471.png">



**- Perfil | Desktop:**
O icone de edição do perfil também não apresente o comportamento de mouseover, acho que poderiamos colocar o mesmo comportamento de mouseover que temos nos botões.

lucca65 commented:
> I honestly can't tell 😬

IIRC that bug was only happening on iOS safari, and I don't have an iPhone I can test with... If you do, please test it out and see if it still happens! I think it should probably be fixed, as I changed a lot how inputs and their borders are treated, but I never figured out what was causing that behavior

I've tested it and its working properly! we can close this one too \o/

juramos-2020 commented:
**- Objective Editor | mobile** 

Ao criar um novo objetivo, acessando o webapp através do google chrome, eu não consigo escrever dentro do campo de texto. Tentei acessar através da aba anônima e consegui escrever, porém ao salvar deu erro "Opa. Algo de errado aconteceu"

- O modal para adicionar o link na palavra aparece com os botões desalinhados e sem espaçamento



**- Objective Editor | Desktop** 

- Erro no editor de texto: Não está sendo possível adicionar um link na palavra
Veja: https://www.loom.com/share/732bb2d6aee447e9be298e9c7581d365

- Também não está sendo possível salvar um novo objetivo na versão desktop;

- Adicionar um espaçamento de 24px entre a linha do "Por favor..." e o "Seu Pin de seis digitos"
![image](https://user-images.githubusercontent.com/67649886/150641543-ee6f6a0a-2ee2-4a49-bd4d-77cef81f91d4.png)

- Na edição de um objetivo: aumentar a largura do botão de "marcar como completo" para não ficar com a linha quebrada.
- Também apresenta o mesmo problema da criação de um novo objetivo: não adiciona link e não salva.


**- Action Editor | mobile** 

- Ao criar uma nova ação, acessando o webapp através do google chrome, eu não consigo escrever dentro do campo de texto. Tentei acessar através da aba anônima e consegui escrever, porém ao salvar deu erro "Opa. Algo de errado aconteceu"

- Ainda na aba anônima, ao tentar anexar um link na palavra, o modal para adicionar o link aparece desconfigurado para a versão mobile, aparece maior que a tela.

- tem vários botões de criar dentro da página de "criar ação"

- ao adicionar um membro validador, esse se repete infinitamente (igual ao erro mensionado anteriormente do botão "criar ação")

- No card da ação as informações relacionadas as recompensas e data estão com cores diferentes. Todas devem ter a cor branca. 

![ação](https://user-images.githubusercontent.com/67649886/150643786-c460bb23-ee45-4722-9d66-1d0e517a05dc.png)

- ao selecionar a opção "validade por data" o campo de "escolha a data" já vem com uma data do ano passado. Acho que deveria puxar a data mais atual, pois isso pode induzir ao erro.

- Também não estou conseguindo salvar a edição de uma ação, aparece o mesmo erro: "Opa. Algo de errado aconteceu".

**- Action Editor | desktop** 

- Não estou conseguindo salvar a criação e a edição de uma ação, aparece o mesmo erro: "Opa. Algo de errado aconteceu".

- Teria como mantermos o padrão de foco dos inputs (esse verdinho com a borda mais grossa) para o text área? Vejo que o foco dessa área de texto é o único que tem a aparência diferente dos demais componentes. 



![ação 1](https://user-images.githubusercontent.com/67649886/150644801-57b84ab7-26a9-4102-8e19-a4137664cf12.png)

juramos-2020 commented:
OI @NeoVier vou aguardar essas alterações, pois acredito que elas vão resolver outros problemas em outras páginas. 😉👍

NeoVier commented:
@juramos-2020 Suspeito que alguns dos problemas que você encontrou são relacionados a extensões instaladas no seu navegador (o Elm, a linguagem que usamos no frontend, pode ter alguns bugs com certas extensões... Existem algumas soluções, mas acho melhor explorarmos essas soluções em outro PR), então poderia listar as extensões que você tem instalado no seu navegador? Assim temos mais dados para testar e descobrir a possível causa desses problemas.

Os problemas da tela de login (onde o PIN é preenchido automaticamente), parecem ser relacionados a um gerenciador de senhas. Você usa algum? Se sim, qual?

Também suspeito que os problemas de duplicação de textos/conteúdo são por conta de extensões, assim como o problema de digitação em campos de rich text.

Para testar esses erros, sugiro testar em aba anônima (ou desabilitar as extensões manualmente)

---

Os problemas de não conseguir salvar objetivos e ações são esperados - fizemos algumas atualizações do contrato, e essa versão do frontend não está atualizada para lidar com essas atualizações. Estamos fazendo isso em #676. Por enquanto, esse erro de "Opa. Algo de errado aconteceu" é esperado.

---

Quanto ao problema de inserir links no rich text, sugiro resolvermos em outro PR também. Já temos a issue #666, que vai mudar algumas coisas no funcionamento interno desse componente, e creio que vai ficar mais fácil de lidar com links.

---

Esses erros visuais foram corrigidos:

> O ícone de "?" do KYC pode seguir a cor padrão dos botões (#F99D33)

> Adicionar um espaçamento de 8px entre o botão "Excluir" do KYC e o texto

> O icone de edição do perfil também não apresente o comportamento de mouseover

> Adicionar um espaçamento de 24px entre a linha do "Por favor..." e o "Seu Pin de seis digitos"

> Na edição de um objetivo: aumentar a largura do botão de "marcar como completo" para não ficar com a linha quebrada.

> No card da ação as informações relacionadas as recompensas e data estão com cores diferentes. Todas devem ter a cor branca.

> ao selecionar a opção "validade por data" o campo de "escolha a data" já vem com uma data do ano passado. Acho que deveria puxar a data mais atual, pois isso pode induzir ao erro.

> Teria como mantermos o padrão de foco dos inputs (esse verdinho com a borda mais grossa) para o text área?

juramos-2020 commented:
OI @NeoVier 

> Os problemas da tela de login (onde o PIN é preenchido automaticamente), parecem ser relacionados a um gerenciador de senhas. Você usa algum? Se sim, qual?

Eu acho que eu não tenho, se eu tenho eu não sei qual é 🙈😅. P.S: na guia anônima esse problema não acontece.

> Para testar esses erros, sugiro testar em aba anônima

Isso, fiz o teste também na guia anônima e de fato esse erro de duplicação de textos/conteúdo e o problema de digitação em campos de rich text, não apareciam nessa guia.

E vai ser incrível se tiver como resolver esse problema das extensões🤩 . O ruim de fazer os testes na guia anônima é que eu não consigo dar print e nem filmar a tela caso precise reportar algum erro. Mas tranquilo eu vou desabilitando as extensões por enquanto. 

Segue as extensões que uso:

Avast Online Security & Privacy 21.0.101

CertiPlugin Chrome 1.12.0

ColorZilla 2.0

Documentos Google off-line 1.39.0

Fonts Ninja 6.0.25

FromDocToPDF (BETA) 13.986.19.62847

Grammarly for Chrome 14.1046.0

Loom – Free Screen and Cam Recorder 5.2.5

Muzli 2 - Stay Inspired 30.2.6

Visual Inspector 1.0.27

WhatFont 2.1.0

Window Resizer 2.6.4

😉👍

NeoVier commented:
> Eu acho que eu não tenho, se eu tenho eu não sei qual é 🙈😅. P.S: na guia anônima esse problema não acontece.

Ok, o fato de não acontecer o erro em guias anônimas é um forte indicativo de que pode ser uma extensão

Obrigado pela lista @juramos-2020, já ajuda muito! Sei que é chato ter que usar a aba anônima, mas creio que podemos deixar para resolver isso em #678 para ir andando com as outras coisas que temos para fazer. @juramos-2020 @lucca65 se concordarem em deixar esses problemas relacionados a extensões para #678, creio que já podemos mergear esse PR, o que acham? Espero a resposta de vocês 🔥

juramos-2020 commented:
@NeoVier 
eu vou fazer um último teste, desta vez todo na aba anônima, pois tiveram alguns fluxos que eu não finalizei o teste. Vou tentar finalizar hoje no mas tardar amanhã sem falta 😉👍

lucca65 commented:
+1, extensões é um daqueles problemas antigos que ainda não tínhamos colocado em issues, obrigado por ter aberto @NeoVier!

por mim resolvemos em separado!

juramos-2020 commented:
OI @NeoVier 

Refiz o teste todo na guia anônima. Segue alguns ajustes:


**Objetivos (versão mobile) -> edição dos objetivos:**
- o botão "MARCAR COMO COMPLETO" deve seguir o mesmo tamanho do botão "SALVAR"
- o modal do PIN (que aparece ao tentar salvar uma edição do objetivo) está desalinhado, está fora da área de visualização mobile com scroll para a lateral.

**Ações (versão mobile) - card das ações**
- os elementos de dentro do card encontram mais alinhados para as esquerda da tela e com pouco espaçamento entre eles. [Segue o design](https://www.figma.com/file/nEPG6Dkns8hDVCudVUTEzc/Dashboard?node-id=2179%3A162)

**Settings: Features (versão mobile e desktop)** 
- os toogles estavam apresentando um comportamento estranho: ao habilitar uma função a outra desabilitava, as vezes eu conseguia habilitar uma função e depois não conseguia mais desabilitar. Não consegui ativar o KYC.
- a palavra "habilitado" do recurso "Patrocinio" não muda de cor quando o toolge é habilitado.
- os dois primeiros toogles funcionam com uma certa lentidão se formos comparar com o comportamento do último toogle.

**Settings: Sponsorship Thank you message (versão mobile e desktop)** 
- não consigo editar o título e nem o campo da mensagem

**Settings: Sponsorship Fiat (versão mobile e desktop)**
- os toogles não estão funcionando


**Settings: News e News Editor (versão mobile e desktop)**
- Quando eu acesso um comunicado para editá-lo e altero a data dele, essa nova data não atualiza no card. Eu salvo a alteração porém a data de agendamento continua sendo a anterior.
- Na versão mobile os componentes de data e horário estão desalinhados. Talvez o campo do horário possa ficar um pouco menor para aumentar o campo da data.
- Ao clicar em "ver mais" é só para o Admin ver como ficou a mensagem dele, não precisa carregar as mensagens anteriores.
- Ao criar um novo comunicado, clicar em "ver mais" e depois clicar na seta de voltar (<) o toogle de "destacar comunicado" é desabilitado automaticamente.
- No dashboard -> na notificação (barra roxa) de novo comunicado: pode deixar um espaçamento de 16px nas bordas laterais, para que o icone do comunicado e o X fiquem mais proximos das bordas laterais.

**Analysis modals/page (versão mobile)**
- Ao abrir um card de uma ação que foi reivindicada, o texto inicial "Esta reivindicação foi cancelada" está muito grande. [Segue o design ](https://www.figma.com/file/LTaoCg6DKv6EPQAlt6Fngs/New-analysis-page?node-id=715%3A4170)

Não entendi qual seria esse fluxo:
-Add contact dashboard ?

Amanhã vou  terminar o testes dos fluxos do shopping 😉👍

juramos-2020 commented:
**Shop Buy (versão mobile)**

- o botão "comprar" está colado no texto "quantas unidades você quer"
- o campo de valor está desabilitado e não é possível completar a compra.

Não entendi qual seria esse fluxo:
-Add contact dashboard ?

NeoVier commented:
- > o botão "MARCAR COMO COMPLETO" deve seguir o mesmo tamanho do botão "SALVAR"

Resolvido ✅ 

- > o modal do PIN (que aparece ao tentar salvar uma edição do objetivo) está desalinhado, está fora da área de visualização mobile com scroll para a lateral.

Não consegui replicar, para mim aparece normal:
<img width="378" alt="image" src="https://user-images.githubusercontent.com/39280468/151187802-9ff8b970-2425-44c8-98fc-2b90007ed676.png">

> os elementos de dentro do card encontram mais alinhados para as esquerda da tela e com pouco espaçamento entre eles

Como esse não é o foco desse PR, apenas ajustei alguns detalhes mais simples, mas sugiro revisitarmos essa tela em outro PR

- > os toogles estavam apresentando um comportamento estranho: ao habilitar uma função a outra desabilitava, as vezes eu conseguia habilitar uma função e depois não conseguia mais desabilitar.

Não consegui replicar, os toggles funcionam normalmente

- > Não consegui ativar o KYC.

Isso é intencional - a comunidade deve falar conosco para habilitar essa funcionalidade (idem para Sponsorship)

- > a palavra "habilitado" do recurso "Patrocinio" não muda de cor quando o toolge é habilitado

Isso acontece porque o toggle está desabilitado (não aceita input) - o mesmo acontece com o KYC, pois esses dois itens devem ser ativados/desativados por nós, manualmente. Talvez possamos mudar o estilo do toggle quando ele está desabilitado, o que acha?

- > os dois primeiros toogles funcionam com uma certa lentidão se formos comparar com o comportamento do último toogle.

Resolvido ✅ 

- > não consigo editar o título e nem o campo da mensagem

Quando estávamos criando o sponsorship, concordamos em deixar assim por enquanto, e se os admins quiserem mudar as configurações, eles devem falar conosco

- > os toogles não estão funcionando

Mesma coisa de cima ☝️ 

- > Quando eu acesso um comunicado para editá-lo e altero a data dele, essa nova data não atualiza no card. Eu salvo a alteração porém a data de agendamento continua sendo a anterior.

Resolvido ✅ 

- > Na versão mobile os componentes de data e horário estão desalinhados. Talvez o campo do horário possa ficar um pouco menor para aumentar o campo da data.

Resolvido ✅ coloquei os campos um em cima o outro em telas menores

- > Ao clicar em "ver mais" é só para o Admin ver como ficou a mensagem dele, não precisa carregar as mensagens anteriores.

Na minha opinião, o fluxo de mostrar na tela de notícias é melhor:
- O admin vê como os outros usuários veem
- Mostrar rich text no espaço limitado do card pode não ser uma boa ideia (pode ser que a notícia tenha um texto com H1, então o texto ficaria muito grande para um espaço muito pequeno - problema maior ainda quando permitirmos imagens no rich text); Acredito que mostrar o texto puro (sem formatações como negrito, itálico, H1, H2, links, imagens, etc.) pode fazer com que o admin ache que a formatação não foi salva.

- > Ao criar um novo comunicado, clicar em "ver mais" e depois clicar na seta de voltar (<) o toogle de "destacar comunicado" é desabilitado automaticamente.

Resolvido ✅ 

- > No dashboard -> na notificação (barra roxa) de novo comunicado: pode deixar um espaçamento de 16px nas bordas laterais, para que o icone do comunicado e o X fiquem mais proximos das bordas laterais.

Resolvido ✅ 

- > Ao abrir um card de uma ação que foi reivindicada, o texto inicial "Esta reivindicação foi cancelada" está muito grande

Resolvido ✅ 

---

> Não entendi qual seria esse fluxo:
Add contact dashboard ?

Esse é o modal de adicionar contato que aparece quando o usuário não tem nenhum contato cadastrado. Na época que implementamos esse modal, decidimos que ele só deveria aparecer por um tempo limitado (até 01/01/2022), então ele não aparece mais. @lucca65 @juramos-2020 acham que devemos mostrar ele ainda ou podemos remover do código?

NeoVier commented:
> o botão "comprar" está colado no texto "quantas unidades você quer"

Resolvido ✅ Coloquei um pouco mais de espaço, mas já vamos revisitar essa tela no shop facelift (#657)

> o campo de valor está desabilitado e não é possível completar a compra.

Esse é o comportamento da tela: o usuário seleciona quantas unidades deseja comprar, e o campo de valor é automaticamente atualizado para mostrar o quanto vai custar a quantidade desejada. Novamente, podemos revisitar melhor esse fluxo no shop facelift

<img width="405" alt="image" src="https://user-images.githubusercontent.com/39280468/151216305-0ce0e8e2-aec8-4de6-9a09-ae1cf7a1f078.png">

lucca65 commented:
> Como esse não é o foco desse PR, apenas ajustei alguns detalhes mais simples, mas sugiro revisitarmos essa tela em outro PR

Temos a #670 que pode ser um bom lugar para resolver e melhorar essa implementação de ações

lucca65 commented:
> Quando estávamos criando o sponsorship, concordamos em deixar assim por enquanto, e se os admins quiserem mudar as configurações, eles devem falar conosco

concordo. @juramos-2020 oq acha de melhorarmos essa comunicação? colocar um elemento para explicar melhor essas situações?

Pode criar uma issue pra isso @juramos-2020? acho que não vai ser complexo de melhorar e vai dar bons resultados

lucca65 commented:
> Esse é o modal de adicionar contato que aparece quando o usuário não tem nenhum contato cadastrado. Na época que implementamos esse modal, decidimos que ele só deveria aparecer por um tempo limitado (até 01/01/2022), então ele não aparece mais. @lucca65 @juramos-2020 acham que devemos mostrar ele ainda ou podemos remover do código?

É verdade, eu tinha esquecido que tinha data pra ele deixar de aparecer.

acho que é útil e deve voltar a existir, ou deveria no mínimo ser incluído diretamente no cadastro, pois muitos usuários ficam incomunicáveis 

Mas também é assunto para outra issue pessoal, acho que por hora podemos colocar o prazo do modal pra essa sexta pra gente testar os inputs e fluxo. assim testamos e tratamos melhor em separado depois.

a próxima versão não sai essa semana então está tranquilo

juramos-2020 commented:
Obrigada @NeoVier  pelos ajustes ficou show !

Sobre o modal do PIN (que aparece ao tentar salvar uma edição do objetivo) e os toogles, eu não consigo te mostrar visualmente porque a guia anônima não permite gravar e nem printar a tela. Mas pelo menos fica o meu registro aqui de como ocorreu o teste 😉👍👍


> Isso acontece porque o toggle está desabilitado (não aceita input) - o mesmo acontece com o KYC, pois esses dois itens devem ser ativados/desativados por nós, manualmente. Talvez possamos mudar o estilo do toggle quando ele está desabilitado, o que acha?

Tranquilo em relação a isso, o que eu quis dizer é que quando habilitamos um toogle a palavra muda de cor. 
Ex: ativei o recurso comunicado -> o toogle fica roxo e a palavra do toogle muda de **desabilitado** para **habilitado** e a cor muda também, ficando roxa. Durante esse teste, em um momento eu consegui ativar o toogle desse recurso e a palavra habilitado continuou cinza. Foi esse o erro que quis reportar. 
Deve ter sido um erro momentâneo, pois testei agora novamente e acho que foi corrigido. O comportamento de não ativar está acontecendo corretamente 😉👍

> Na minha opinião, o fluxo de mostrar na tela de notícias é melhor:

sobre isso, eu quis dizer com `Ao clicar em "ver mais" é só para o Admin vê como ficou a mensagem dele, não precisa carregar as mensagens anteriores.` -> É que quando o admin clica em **"ver mais"** ele vê apenas o comunicado que ele acabou de criar, ele não precisa ver aquela listagem das mensagens criadas anteriormente. [Segue design](https://www.figma.com/file/De9z2EjpGRkfsEXVzfj9u2/Support-News-%26-Digest-E-mails?node-id=471%3A5185).

juramos-2020 commented:
> É verdade, eu tinha esquecido que tinha data pra ele deixar de aparecer.
acho que é útil e deve voltar a existir, ou deveria no mínimo ser incluído diretamente no cadastro, pois muitos usuários ficam incomunicáveis

+1

juramos-2020 commented:
> Quando estávamos criando o sponsorship, concordamos em deixar assim por enquanto, e se os admins quiserem mudar as configurações, eles devem falar conosco

> concordo. @juramos-2020 oq acha de melhorarmos essa comunicação? colocar um elemento para explicar melhor essas situações?

Pensei e colocar o mesmo elemento que tem no KYC, aquele "?" com uma mensagem: Para ativar esse recurso a pessoa que administra a comunidade deve entrar em contato com equipe da Cambiatus para solicitar essa ativação. 

O que vocês acham?

juramos-2020 commented:
@lucca65  @NeoVier [Nova issue criada ](https://github.com/cambiatus/frontend/issues/679)  😉👍

NeoVier commented:
@lucca65 @juramos-2020 aumentei a data limite do modal para essa sexta, agora é só acessar o dashboard sem nenhum contato cadastrado que ele aparece.

> sobre isso, eu quis dizer com Ao clicar em "ver mais" é só para o Admin vê como ficou a mensagem dele, não precisa carregar as mensagens anteriores. -> É que quando o admin clica em "ver mais" ele vê apenas o comunicado que ele acabou de criar, ele não precisa ver aquela listagem das mensagens criadas anteriormente

Entendi. Resolvido ✅

NeoVier commented:
@lucca65 então acho que estamos prontos para mergear! Só preciso de uma aprovação no PR
